File: rb/features.rb
# TransportrestTransitApis SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module TransportrestTransitApisFeatures
  def self.make_feature(name)
    case name
    when "base"
      TransportrestTransitApisBaseFeature.new
    when "ratelimit"
      TransportrestTransitApisRatelimitFeature.new
    when "retry"
      TransportrestTransitApisRetryFeature.new
    when "test"
      TransportrestTransitApisTestFeature.new
    when "timeout"
      TransportrestTransitApisTimeoutFeature.new
    else
      TransportrestTransitApisBaseFeature.new
    end
  end
end

File: py/features.py
# TransportrestTransitApis SDK feature factory

from feature.base_feature import TransportrestTransitApisBaseFeature
from feature.test_feature import TransportrestTransitApisTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TransportrestTransitApisBaseFeature(),
        "test": lambda: TransportrestTransitApisTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
